extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -257,6 +273,45 @@ export class SubscriptionService {
     return this;
   }
 
+  /**
+   * Delegates authentication to viewer
+   */
+  delegateAuthToViewer_() {
+    const serviceIds = ['local'];
+    const currentProductId = /** @type {string} */ (user().assert(
+        this.pageConfig_.getProductId(),
+        'Product id is null'
+    ));
+    const publicationId = /** @type {string} */ (user().assert(
+        this.pageConfig_.getPublicationId(),
+        'Publication id is null'
+    ));
+    const origin = getWinOrigin(this.ampdoc_.win);
+    this.platformStore_ = new PlatformStore(serviceIds);
+    this.viewer_.sendMessageAwaitResponse('auth', dict({
+      'publicationId': publicationId,
+      'productId': currentProductId,
+      'origin': origin,
+    })).then(entitlementData => {
+      const authData = (entitlementData || {})['authorization'];
+      dev().assert(authData, 'authorization is not defined');
+      const decodedData = this.jwtHelper_.decode(authData);
+      if (decodedData['aud'] != origin) {
+        user().error(TAG, 'Audience does not match');
+      }
+      if (Date.now() - decodedData['exp'] < 1000 * 60) {
+        user().error(TAG, 'Payload about to expire');
+      }
+      const entitlements = decodedData['entitlements'];
+      user().assert(entitlements[0], 'No entitlements found');
```

Comment:
To make things more complicated (though TODO + next PR is fine):

1. It could be either an object == single entitlement, or an array.
2. When an array, we want to cycle over it, parse each item and break when we found the first matching.
3. It's totally fine to have no `entitlements` field or an empty array -- that indicates no permissions (in other words `Entitlement.empty()`).

Reply:
Done